meta-client/src/main/java/org/apache/helix/metaclient/recipes/leaderelection/LeaderElectionClient.java

```diff
@@ -0,0 +1,125 @@
+package org.apache.helix.metaclient.recipes.leaderelection;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.List;
+
+import org.apache.helix.metaclient.api.MetaClientInterface;
+import org.apache.helix.metaclient.factories.MetaClientConfig;
+
+
+/**
+ * LeaderElectionClient does distributed leader election using CRUD and change notification APIs
+ * provided by underlying metadata client. Leader election config can provide many
+ * configs like base path for all participating nodes, sync/async mode, TTL etc.
+ *
+ * Participants join a leader election group by calling the following API.
+ * The Leader Election client maintains and elect an active leader from participant pool
+ * All participants wanted to be elected as leader joins a pool
+ * LeaderElection client maintains an active leader, by monitoring liveness of current leader and
+ * re-elect if needed and user no need to call elect or re-elect explicitly.
+ * This LeaderElection client will notify registered listeners for any leadership change.
+ */
+public class LeaderElectionClient {
```

Comment:
One question for my understanding, do we want the `LeaderElectionClient` to be for a specific `path` or is it like this is one single stop for all the leadership related stuff for multiple paths. I feel that we should make it specific to a path, and sort of have a manager that manages multiple paths if wanted. In this way we keep the `LeaderElectionClient` clean. 

Can you also add that to the description if possible.

Reply:
Thanks for raising the question. 
This client is designed to be used for 
1. End user that need to leader ship election or coordination. There may be one leader election group or more than one leader election group on each app or JVM.
2. Potentially an leader election service that hosts distributed leader election service.

In either case, more than one path may co-exist. Having one client is more efficient.

Reply:
Gotcha! I agree that we should have one client. It's more from a naming perspective where I feel there should a `LeaderElectionManager` that manages `LeaderElectionClient which only deals with one single path` and consumers interact with the manager and not the client.

Reply:
`LeaderElectionManager ` sounds like a centralized service. It is still a client.

Reply:
Ok. Thanks!